src/cryptography/verify_signature.py

```diff
@@ -2,29 +2,50 @@
 from typing import Any, List, Tuple
 
 from eth_account import Account
+from eth_account.account import VRS
 from web3 import Web3
 
 logger = logging.getLogger(__name__)
 
+V_OFFSET = 27
+
+
+# Solidity function
+#
+# function recover(bytes32 hash, bytes32 r, bytes32 vs) internal pure returns (address) {
+#        bytes32 s;
+#        uint8 v;
+#        assembly {
+#            s := and(vs, 0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)
+#            v := add(shr(255, vs), 27)
+#        }
+#        return recover(hash, v, r, s);
+#    }
+def recover_vs(vs: str) -> tuple[VRS, VRS]:
+    """
+    Recovers v and s parameters of the signature from _vs field
+    """
+    # cut 0x
+    _vs = int.from_bytes(bytearray.fromhex(vs[2:]))
+    s = _vs & 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+    v = (_vs >> 255) + V_OFFSET
+    return v, s
+
 
 def compute_vs(v: int, s: str) -> str:
     """Returns aggregated _vs value."""
-    if v < 27:
-        if v in [0, 1]:
-            v += 27
-        else:
-            msg = 'Signature invalid v byte.'
-            logger.error({'msg': 'Signature invalid v byte.', 'data': str(v)})
-            raise ValueError(msg)
-
+    if v < V_OFFSET and v not in [0, 1]:
+        raise ValueError('Signature invalid v byte.')
```

Comment:
Lets log `v` variable, like in previous version

Reply:
Done